src/fides/api/ops/api/v1/endpoints/consent_request_endpoints.py

```diff
@@ -502,22 +479,37 @@ def _get_consent_request_and_provided_identity(
 
 
 def _prepare_consent_preferences(
-    db: Session, provided_identity: ProvidedIdentity
+    db: Session,
+    provided_identity: ProvidedIdentity,
+    consents: Optional[List[ConsentSchema]] = None,
 ) -> ConsentPreferences:
-    consent = Consent.filter(
+    """
+    Find all of the consents an identity has.
+
+    If a list of `consents` is passed in, will also match and return which Consents are `executable`.
+    Otherwise, `executable` will be set to None.
+    """
+    consents_matching_identity: List[Consent] = Consent.filter(
         db=db, conditions=Consent.provided_identity_id == provided_identity.id
     ).all()
 
-    if not consent:
+    if not consents_matching_identity:
         return ConsentPreferences(consent=None)
 
-    return ConsentPreferences(
-        consent=[
+    consent_list = []
+    for consent in consents_matching_identity:
```

Comment:
this is the part I'm most unsure about. it seems like we want to merge what the client gives us (which has `executable`) with the Consent objects from the db (which don't have `executable`). it feels a lil weird to me, so I would love some feedback!

Reply:
I agree, let's try to adjust to touch less code. Maybe https://github.com/ethyca/fides/pull/2193/files#r1067069764